system/lib/libc/musl/src/signal/setitimer.c

```diff
@@ -37,11 +37,15 @@ void _emscripten_timeout(int which, double now)
 		signum = SIGPROF;
 	else if (which == ITIMER_VIRTUAL)
 		signum = SIGVTALRM;
-	int next_timeout = current_intervals_ms[which];
-	if (next_timeout)
-		current_timeout_ms[which] = now + next_timeout;
-	else
+	double next_timeout = 0.0;
```

Comment:
This worries me slightly since if we end up with a fraction that is close to 0, rounding it down to 0 would be a problem (since 0 is special). I don't think it will actually be a problem in the current code, but if we changed the signature of `_setitimer_js` from double to int, then it could be, so that seems worth a comment maybe.

Reply:
This still represents a number of milliseconds, so there should never actually be a fraction. I switched it to a double because all the existing variables representing time are also doubles (including emscripten_get_now(), and of course in JS). I think what's best is either the current way, (or maybe switching  all the variables here to uint/uint64).

Reply:
Ah, I see, makes sense. This part lgtm as is then.